pandas/tests/sparse/test_frame.py

```diff
@@ -540,6 +540,37 @@ def test_setitem_array(self):
                                   self.frame['F'].reindex(index),
                                   check_names=False)
 
+    def test_setitem_spmatrix(self):
+        # GH-15634
+        tm.skip_if_no_package('scipy')
+        from scipy.sparse import csr_matrix
+
+        sdf = self.frame.copy(False)
+
+        def _equal(spm1, spm2):
+            return np.all(spm1.toarray() == spm2.toarray())
+
+        # 1d -- column
+        spm = csr_matrix(np.arange(len(sdf))).T
+        sdf['X'] = spm
+        assert _equal(sdf[['X']].to_coo(), spm)
+
+        # 1d -- existing column
+        sdf['A'] = spm.T
+        assert _equal(sdf[['X']].to_coo(), spm)
+
+        # 1d row -- changing series contents not yet supported
+        spm = csr_matrix(np.arange(sdf.shape[1], dtype=float))
+        idx = np.zeros(sdf.shape[0], dtype=bool)
```

Comment:
can you test with ``.loc/.iloc`` as well (might already be another issue about this, if not and its too complicated for here, then create a new issue)